### PR TITLE
overlord/snapstate: pick up system defaults when seeding the snapd snap

### PR DESCRIPTION
**Pull request by bboozzoo**
On a system with bases, the snapd snap is considered a 'system' snap, thus we
consider it as source of system configuration defaults. Capture this in the
code, make sure that system defaults are only applied when in the seeding mode.
Distinguish cases when both snapd and core snaps are installed, or just one of
those.

### CODE REVIEW

**Comment by pedronis** [comment]
Looking at this again, is not super clear what the bug is about, we should expand tests/main/ubuntu-core-gadget-config-defaults to cover "system" and core18.

I'm not entirely sure the new code works well for a classic device with snapd but a gadget that has "core" as its base.

File: overlord/configstate/hooks.go [hunk 1]
```diff
@@ -96,7 +96,8 @@ func (h *configureHandler) Before() error {
 		if err != nil && err != state.ErrNoState {
 			return err
 		}
-		if len(patch) != 0 {
+		// core does not need a configure hook
```
Comment (pedronis) [inline overlord/configstate/hooks.go]: // core is handled internally and does not need a configure hook

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -2347,18 +2347,20 @@ func ConfigDefaults(st *state.State, deviceCtx DeviceContext, snapName string) (
 		return nil, err
 	}
 
+	isSystemDefaults := snapName == defaultCoreSnapName
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: // system configuration is kept under "core" so apply it's defaults when configuring "core"

Comment (pedronis) [reply overlord/snapstate/snapstate.go]: @bboozzoo  oops s/it's/its/

File: overlord/configstate/hooks.go (hunk 1 above)
Comment (bboozzoo) [reply overlord/configstate/hooks.go]: Missed that when I was pushing the updates. I'll fix this once the PR lands on master.

**Pull request merged**